### PR TITLE
release: v2.4.7 and properly sync module versions

### PR DESCRIPTION
**Pull request by kruskall**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -84,6 +84,7 @@ For example, APM Server version `7.15.0` corresponds with the Go APM Agent versi
 
 1. Update `version.go` and `internal/apmversion/version.go`, and then run `make update-modules`
 2. Update [`CHANGELOG.asciidoc`](changelog.asciidoc), by adding a new version heading (`==== 1.x.x - yyyy/MM/dd`) and changing the base tag of the Unreleased comparison URL
+3. Run `make update-modules`to sync the dependencies version
```
Comment (carsonip) [inline CONTRIBUTING.md:87]:
```suggestion
3. Run `make update-modules` to sync the dependencies version
```

File: CHANGELOG.asciidoc [hunk 2]
```diff
@@ -21,7 +21,7 @@ endif::[]
 [[unreleased]]
 === Unreleased
 
-https://github.com/elastic/apm-agent-go/compare/v2.4.6...main[View commits]
+https://github.com/elastic/apm-agent-go/compare/v2.4.7...main[View commits]
```
Comment (carsonip) [inline CHANGELOG.asciidoc:24]: add changelog

Comment (axw) [reply CHANGELOG.asciidoc:24]: Perhaps add a note that there was an issue in the module tags for v2.4.6, and that's why there's v2.4.7?

File: CHANGELOG.asciidoc [hunk 3]
```diff
@@ -21,11 +21,16 @@ endif::[]
...
 
 [[release-notes-2.x]]
 === Go Agent version 2.x
 
+[[release-notes-2.4.7]]
+==== 2.4.7 - 2023/11/23
+
+- Bump submodule dependency version {pull}1546[#1546]
```
Comment (carsonip) [inline CHANGELOG.asciidoc:32]: nit: if there was an issue with 2.4.6 I'd prefer that we mention it, either under 2.4.7 or 2.4.6.

**Review by carsonip — APPROVED** [review]

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -84,6 +84,7 @@ For example, APM Server version `7.15.0` corresponds with the Go APM Agent versi
 
 1. Update `version.go` and `internal/apmversion/version.go`, and then run `make update-modules`
 2. Update [`CHANGELOG.asciidoc`](changelog.asciidoc), by adding a new version heading (`==== 1.x.x - yyyy/MM/dd`) and changing the base tag of the Unreleased comparison URL
+3. Run `make update-modules` to sync the dependencies version
```
Comment (axw) [inline CONTRIBUTING.md:87]: I'm a bit late, but this is already mentioned at the end of step 1. Perhaps split it into a following line if it's easy to miss.

**Pull request merged**